src/middleware/error-handler.js

```diff
@@ -1,16 +1,21 @@
-const logger = require('../utils/logger');
+const { logger } = require("../utils/logger");
 
-const errorHandler = (
-  err,
-  req,
-  res,
-  next
-) => {
-  logger.error(err.message);
-  if (!err.statusCode) err.statusCode = 500;
-  res.status(err.statusCode).send(err.message);
+const ERROR_CODES = {
+  UNPROCESSABLE_ENTITY: 422,
+  NOT_FOUND: 404,
+  INTERNAL_SERVER_ERROR: 500
+};
```

Comment:
Very good! Adding a label to these numeric codes makes it much easier to understand their meaning for someone who hasn't memorized the HTTP spec.

One tiny change I'd suggest it to name the containing object 'HTTP_STATUS_CODES` or `HTTP_STATUS` or something similar. You may eventually want to also include non-error status codes in this object (e.g. 200 OK, 201 CREATED, etc) and move it into a utils module or something and import it here.  You don't have to do this now, but it's something to keep in mind.  Alternatively, you may decide to have multiple http status code objects (e.g. one for errors, one for success etc), in which case this would work.

Reply:
I'll add a separate object for success codes. Thanks for the recommendation! Makes total sense, not sure why I didn't think of it!